e2e/jest.test-setup.ts

```diff
@@ -128,45 +152,69 @@ beforeEach(async () => {
   });
 
   page.on('requestfinished', async (request) => {
-    if (canLogResponse(request)) {
-      if (isApiFailure(request)) {
-        await logError(request);
-      } else {
-        await logResponse(request);
+    let method;
+    let url;
+    let status;
+    try {
+      if (canLogResponse(request)) {
+        // Try find out what the request was if exception thrown.
+        method = request.method();
+        const resp = request.response();
+        url = resp.url();
+        status = resp.status();
+
+        if (isApiFailure(request)) {
+          await logError(request);
+        } else {
+          const responseBody = await transformResponseBody(request);
+          if (skipApiResponseBody(request)) {
+            console.debug('❗ Request finished: ' +
+               `${request.response().status()} ${request.method()} ${request.url()}`);
+          } else {
+            console.debug('❗ Request finished: ' +
+               `${request.response().status()} ${request.method()} ${request.url()}\n${responseBody}`);
+          }
+        }
       }
+    } catch (err) {
+      // Try find out what the request was
+      console.error(`${err}\n${status} ${method} ${url}`);
     }
     try {
       await request.continue();
-      // tslint:disable-next-line:no-empty
     } catch (e) {
+      // Ignored
     }
   });
 
   page.on('console', async (message) => {
     if (!message.args().length) {
       return;
     }
+    const title = await getTitle();
     try {
-      const title = await getTitle();
-      const args = await Promise.all(message.args().map(a => a.jsonValue()));
-      console[message.type() === 'warning' ? 'warn' : message.type()](`❗ ${title}\n`, ...args);
-      // tslint:disable-next-line:no-empty
+      const args = await Promise.all(message.args().map(a => describeJsHandle(a)));
+      console[message.type() === 'warning' ? 'warn' : message.type()](`❗ ${title}\n${message.text()}`, ...args);
     } catch (err) {
-      console.error(`Exception occurred when getting console messages.\n${err}`);
+      console.error(`❗ ${title}\nException occurred when getting console message.\n${err}\n${message.text()}`);
     }
   });
 
   page.on('error', async (error) => {
     const title = await getTitle();
-    console.error(`❗ ${title}\npage error: ${error}`);
+    try {
+      console.error(`❗ ${title}\nError message: ${error.message}\nStack: ${error.stack}`);
+    } catch (err) {
+      console.error(`❗ ${title}\nException occurred when getting page error.\n${err}`);
+    }
   });
 
   page.on('pageerror', async (error) => {
     const title = await getTitle();
     try {
-      console.error(`❗ ${title}\npage error: ${error}`);
-      // tslint:disable-next-line:no-empty
+      console.error(`❗ ${title}\nError message: ${error.message}\nStack: ${error.stack}`);
```

Comment:
nit: to distnguish this from the page.on('error') handler, maybe add a "Page" prefix:

`! ${title}\nPage error message: ...

Reply:
👍